src/main.ts

```diff
@@ -2,13 +2,29 @@ import 'source-map-support/register';
 import { logger } from './modules/common/Logger';
 import Stoa from './Stoa';
 
+import { ArgumentParser } from "argparse";
 import express from 'express';
+import shell from 'shelljs';
 import { URL } from 'url';
 
-const port: number = Number(process.env.PORT) || 3836;
-const agora_address: URL = new URL(process.env.AGORA_ENDPOINT || "http://127.0.0.1:2826");
-// TODO: Replace hardcoded "127.0.0.1" with  agora_address.hostname (#106)
-const stoa: express.Application = new Stoa("database", "127.0.0.1", agora_address.port).stoa;
+// Compile JSON Schema to TypeScript Type
+shell.exec("npm run json2ts");
+
+// Parse the arguments
+const parser = new ArgumentParser();
+parser.add_argument('-a', '--agora', { help: "The endpoint of Agora" });
+parser.add_argument('-p', '--port', { help: "The network port of Stoa" });
```

Comment:
This is good too.
It would be nice if I could parse this to the address.
It would be better to enter only one optional parameter.

Reply:
I'd appreciate it if you could be more specific.

Reply:
For example, we would like to support `--agora http://127.0.0.1:2826` as the parameter address value.

Reply:
Ah, this seems so right.